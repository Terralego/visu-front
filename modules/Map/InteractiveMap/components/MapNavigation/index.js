import React from 'react';
import MapNavigation from './MapNavigation';
import { connectTerraFrontProvider } from '../../../../TerraFrontProvider';

export * from './MapNavigation';
export default connectTerraFrontProvider({
  renderMapNavigation: 'modules.Visualizer.widgets.Map.renderMapNavigation',
})(
  ({
    renderMapNavigation: RenderMapNavigation = MapNavigation,
    ...props
  }) => <RenderMapNavigation defaultRender={MapNavigation} {...props} />,
);
