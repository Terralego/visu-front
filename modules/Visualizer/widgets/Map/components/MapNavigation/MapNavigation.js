import React from 'react';
import { Button, Card, Classes, Tooltip } from '@blueprintjs/core';

import LayersTree from '../LayersTree';

export const MapNavigation = ({
  toggleLabel,
  onToggle,
  isVisible,
  ContentComponent,
  ...props
}) => (
  <Card
    className={`map-navigation ${Classes.DARK}`}
  >
    <Tooltip
      className="map-navigation__button-container"
      content={isVisible ? 'replier ' : 'dÃ©plier'}
    >
      <Button
        className="map-navigation__button"
        onClick={onToggle}
        icon="arrow-right"
        minimal
      />
    </Tooltip>
    <div className="map-navigation__content">
      <ContentComponent
        defaultContentComponent={MapNavigation.defaultProps.ContentComponent}
        {...props}
      />
    </div>
  </Card>
);
MapNavigation.defaultProps = {
  onToggle () {},
  ContentComponent (props) {
    return <LayersTree {...props} />;
  },
};

export default MapNavigation;
