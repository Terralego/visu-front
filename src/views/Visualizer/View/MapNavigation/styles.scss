@import '../../../variables.scss'; 

$transitionSpeed: .2s;

.map-navigation {
  position: relative;
  height: 100vh;
  width: 19.5rem;
  margin-left: -19.5rem;
  transition: margin $transitionSpeed ease-in;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  flex-shrink: 0;
  border-radius: 0;
  padding: 0 0.7rem;
  z-index: 80;
  box-shadow: 8px 0 8px 8px rgba(16,22,26,0.20), 9px 8px 24px 0 rgba(16,22,26,0.40);
  
  &__header {    
    opacity: 1;
    width: 100%;
    transition: opacity $transitionSpeed ease-in;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    background-image: url('../../../../images/visu-background.png');
    height: 3rem;
  }
  &__title {
    padding-left: .5rem;
    margin-bottom: .5rem;
    margin-top: .25rem;
    font-weight: 300;
    letter-spacing: 1.25px;
    font-size: 1.2rem;
  }

  &__button-container {
    margin-left: auto;
    position: absolute;
    right: -2rem;
  }

  &__button {
    width: 1.5rem;
    height: 1.5rem;
    border-radius: 0;
    margin: .5rem 0;
    outline: none;
    transition: transform $transitionSpeed ease-in;
    transform: translate3d(8px, 0, 0);
  }

  &__button-tooltip {
    z-index: 120;
    position: relative;
  }

  &__content {
    opacity: 0;
    width: 100%;
    transition: opacity $transitionSpeed ease-in;
    overflow-y: auto;
    display: flex;
    flex-direction: row;
  }
  &__content::-webkit-scrollbar {
    width: 8px;
  }

  @media print {
    display: none;
  }
}

.visualizer-view {
  &.is-layers-tree-visible {

    .map-navigation {
      margin-left: 0;
      padding: 0 0.7rem;
  
      &__button {
        transform: scaleX(-1) translate3d(-0.5rem, 0, 0);
      }
  
      &__content {
        opacity: 1;
      }
    }
  }
}

// Styles relative to DarkTheme
.bp3-dark {

  .map-navigation__content {
    scrollbar-color: #202B33 #30404D;
  }

  .map-navigation__content::-webkit-scrollbar-track {
    background-color: #30404D;
  }
  
  .map-navigation__content::-webkit-scrollbar-thumb {
    background-color: #202B33;
  }
}

@media(min-width: $breakpoints-tablet) and (max-width: $breakpoints-desktop) and (orientation: landscape) {

  .map-navigation {
    width: 17.5rem;
    margin-left: -17.5rem;
    transition: margin $transitionSpeed ease-in;
  
    &__title {
      margin-top: .5rem / $scaleRatioTablet;
      font-size: 1.2rem / $scaleRatioTablet;
    }
  
    &__button {
      width: 1.5rem / $scaleRatioTablet;
      height: 1.5rem / $scaleRatioTablet;
      border-radius: 0;
      margin: .5rem 0 / $scaleRatioTablet;
      transform: translate3d((.5rem / $scaleRatioTablet), 0, 0);
    }

    &__content {
      .layerstree-panel-list {
        padding: 0 .25rem 0 .5rem;
      }
    }

    &__content::-webkit-scrollbar {
      width: .5rem / $scaleRatioTablet;
    }
  }
}

@media(min-width: $breakpoints-phone) and (max-width: $breakpoints-tablet) and (orientation: portrait) {
  .visualizer-view {
    &.is-layers-tree-visible {
      .map-navigation__button {
        transform: scaleX(-1) translate3d(2.4rem, 0, 0);
      }
    }
  }
  .map-navigation {
    width: 14rem;
    margin-left: -14rem;

    &__header {
      height: 4rem;
    }
  
    &__title {
      font-size: 1.2rem / $scaleRatioTablet;
    }
    &__content {
      .layerstree-panel-list {
        padding: 0 .25rem 0 .5rem;
      }
    }
    &__content::-webkit-scrollbar {
      width: .5rem / $scaleRatioTablet;
    }
  }
}