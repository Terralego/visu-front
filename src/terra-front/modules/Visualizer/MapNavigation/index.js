import MapNavigation from './MapNavigation';

export default MapNavigation;
