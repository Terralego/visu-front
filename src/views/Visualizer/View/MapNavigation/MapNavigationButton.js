import React from 'react';
import { Button, Tooltip } from '@blueprintjs/core';

import './styles.scss';

export default ({ isVisible, onToggle, uid }) => (
  <Tooltip
    className="bp3-dark map-navigation__button-container"
    portalClassName="map-navigation__button-tooltip"
    content={isVisible ? 'replier ' : 'dÃ©plier'}
  >
    <Button
      className="map-navigation__button"
      onClick={onToggle}
      aria-controls={`map-navigations__content-${uid}`}
      aria-expanded={isVisible}
      icon="arrow-right"
    />
  </Tooltip>
);
