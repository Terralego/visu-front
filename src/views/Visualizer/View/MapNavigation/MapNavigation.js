import React from 'react';
import { Card, Classes } from '@blueprintjs/core';
import { LayersTree } from '@terralego/core/modules/Visualizer';

import MapNavigationButton from './MapNavigationButton';

import './styles.scss';

function getUid () {
  return Math.floor((Date.now() * Math.random())).toString(16);
}

export const MapNavigation = ({
  title,
  children,
  visible,
  toggleLayersTree,
  renderHeader,
}) => {
  const uid = getUid();
  return (
    <Card
      className={`map-navigation ${Classes.DARK}`}
    >
      {renderHeader && (
        <div className="map-navigation__header">
          {renderHeader}
        </div>
      )}
      {title && <h2 className="map-navigation__title">{title}</h2>}
      <div id={`map-navigation__content-${uid}`} className="map-navigation__content">
        {children}
      </div>
      <MapNavigationButton
        onToggle={toggleLayersTree}
        isVisible={visible}
      />
    </Card>
  );
};
MapNavigation.defaultProps = {
  onToggle () {},
  ContentComponent (props) {
    return <LayersTree {...props} />;
  },
};

export default MapNavigation;
