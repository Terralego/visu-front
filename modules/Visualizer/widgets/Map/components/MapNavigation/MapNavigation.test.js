import React from 'react';
import renderer from 'react-test-renderer';
import MapNavigation from './MapNavigation';

jest.mock('@blueprintjs/core', () => ({
  Card ({ children }) {
    return children;
  },
  Button () {
    return 'LayerTree Button';
  },
  Tooltip  ({ children }) {
    return children;
  },
  Classes: { DARK: 'dark' },
}));
jest.mock('../LayersTree', () => function LayersTree () {
  return <p>LayersTree</p>;
});

it('should render correctly', () => {
  const tree = renderer.create((
    <>
      <MapNavigation />
      <MapNavigation isVisible />
    </>
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
