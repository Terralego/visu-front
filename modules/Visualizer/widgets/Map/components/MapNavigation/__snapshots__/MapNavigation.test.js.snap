// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
Array [
  "LayerTree Button",
  <div
    className="map-navigation__content"
  >
    <p>
      LayersTree
    </p>
  </div>,
  "LayerTree Button",
  <div
    className="map-navigation__content"
  >
    <p>
      LayersTree
    </p>
  </div>,
]
`;
